edn/src/types.rs

```diff
@@ -398,6 +398,12 @@ macro_rules! def_common_value_impl {
             def_common_value_methods!($t<$tchild>);
         }
 
+        impl AsRef<$t> for $t {
```

Comment:
I like not necessarily having to make the borrow explicit, and pattern matching should be able to work on both owned or borrowed values. So, just as pretty much any other use of AsRef, convenience, in this case just abstracting over the fact that borrowing is an implementation detail. Can remove if you don't like it.